src/main/java/functions/AnalyseCommand.java

```diff
@@ -0,0 +1,71 @@
+package functions;
+
+public class AnalyseCommand {
+
+    protected String commandType;
+    protected String commandDescription;
+    protected String commandTime;
+
+    public AnalyseCommand(String input) {
+        setCommandType(input);
+        setCommandDescription(input);
+        setCommandTime(input);
+
+    }
+
+    public void setCommandType(String input) {
+        if (input.contains(" ")) {
+            String[] commandType = input.split(" ");
+            this.commandType = commandType[0].toLowerCase();
+        } else {
+            this.commandType = input.toLowerCase();
+        }
+    }
+
+    public String getCommandType(){
+        return this.commandType;
+    }
+
+    public void setCommandDescription(String input) {
+        if (input.contains(" ")) {
+            String[] commandDescription = input.split(" ", 2);
+            if (commandDescription[0].equalsIgnoreCase("deadline") || commandDescription[0].equalsIgnoreCase("event")) {
```

Comment:
Should this be split out into methods instead? It might make the code more readable?
For eg maybe a function named userEnteredDeadline or userEnteredEvent